### PR TITLE
MicroscopyModelFactory: code cleaning

### PR DESCRIPTION
**Pull request by annagav**

### CODE REVIEW

File: html_app/components/MicroscopyModelFactory.js [hunk 1]
```diff
@@ -67,45 +64,33 @@ scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelF
...
-									isFound = true;
-								}
-								else{
+                                    number_of_comparisons = 0;
+                                    do{
+                                        /*Math.random returns num in [0,1) so no need to worry about subtracting 1 from index*/
+                                        index =  Math.floor(Math.random() * (max));
+                                        slide_array = template.slide_parser[collection_id][slide_type][conditions][index];
+                                        alreadySelected = false;
+                                        _.each(slide_array, function(x){if(_.contains(hash_list, x.hash)) alreadySelected=true;});
```
Comment (carsongee) [inline html_app/components/MicroscopyModelFactory.js]: Break into multiple lines

File: html_app/components/MicroscopyModelFactory.js [hunk 2]
```diff
@@ -67,45 +64,33 @@ scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelF
...
+                                        number_of_comparisons += 1;
+                                    }while(alreadySelected && number_of_comparisons < max);
+                                    imgs=slide_array;
+                                    isFound = true;
+                                }else{
```
Comment (carsongee) [inline html_app/components/MicroscopyModelFactory.js:82]: align ifs and elses, tabs and spaces again?

File: html_app/components/MicroscopyModelFactory.js [hunk 3]
```diff
@@ -67,45 +64,33 @@ scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelF
...
 				}
 			else if (parser.complex_parser){
 					var micro_state = {
-						drug_id: function (arr) {
-							var hasVal = false;
-							for(var x = 0; x < arr.length; x++){
-								if(arr[x] == drug_id)
-									hasVal = true;
-							}
-							return hasVal;
```
Comment (carsongee) [inline html_app/components/MicroscopyModelFactory.js:108]: :sparkles: :sparkling_heart:

**Comment by annagav** [comment]
Aligned if-else. Restructured for-each loop.

File: html_app/components/MicroscopyModelFactory.js [hunk 4]
```diff
@@ -67,45 +64,36 @@ scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelF
...
-								}
-								else{
+                                    number_of_comparisons = 0;
+                                    do{
+                                        /*Math.random returns num in [0,1) so no need to worry about subtracting 1 from index*/
+                                        index =  Math.floor(Math.random() * (max));
+                                        slide_array = template.slide_parser[collection_id][slide_type][conditions][index];
+                                        alreadySelected = false;
+                                        _.each(slide_array, function(x){
+                                            if(_.contains(hash_list, x.hash))
```
Comment (carsongee) [inline html_app/components/MicroscopyModelFactory.js]: can you add braces here? I know they aren't strictly required for one line ifs, but it is much clearer if they are.

**Comment by carsongee** [comment]
Other than that that last little one, should be merge ready.

**Comment by annagav** [comment]
Added the 'if' braces.

**Pull request merged**